pennylane_lightning_kokkos/_serialize.py

```diff
@@ -209,12 +200,16 @@ def _serialize_ob(ob, wires_map, use_csingle):
         return _serialize_hamiltonian(ob, wires_map, use_csingle)
     elif ob.name == "SparseHamiltonian":
         return _serialize_sparsehamiltonian(ob, wires_map, use_csingle)
+    elif isinstance(ob, (PauliX, PauliY, PauliZ, Identity, Hadamard)):
+        return _serialize_named_ob(ob, wires_map, use_csingle)
+    elif ob._pauli_rep is not None:
+        return _serialize_pauli_sentence(ob._pauli_rep, wires_map, use_csingle)
     elif ob.name == "Hermitian":
         raise TypeError(
-            f"Hermitian observables are not currently supported for adjoint differentiation. Please use Pauli-words only."
+            "Hermitian observables are not currently supported for adjoint differentiation. Please use Pauli-words only."
         )
     else:
-        return _serialize_named_ob(ob, wires_map, use_csingle)
+        raise TypeError(f"Unknown observable found: {ob}. Please use Pauli-words only.")
```

Comment:
the key difference from the lightning change ([those lines](https://github.com/PennyLaneAI/pennylane-lightning/blob/769997cdce4096b933129d421fc703dc9d728f37/pennylane_lightning/_serialize.py#L157-L158), for reference) is in these lines.

In pennylane-lightning, if an observable isn't a Hamiltonian, Tensor, Named, or something with a pauli rep, it just uses `Hermitian` which calls `qml.matrix`.

Here, if it doesn't meet any of the other requirements, it just raises an error. Lightning uses `_serialize_ob` even for regular expval, but this repo (and GPU) only serializes obs for adjoint differentiation, which I guess the C++ hermitian class doesn't support?

I gave a new error if the observable isn't support and it's not a `qml.Hermitian`, but we could combine them if we want.

Reply:
I'll take a quick look at this --- Hermitian should be supported here in the adjoint execution pipeline, though only for non-parametric gate evaluation.

Reply:
Just checked, this shouldn't be disabled. I'll make a PR that fixes this --- we can then replicate the same dispatch function as with lightning.qubit and remove the error.